cluster/v1alpha1/helpers.go

```diff
@@ -236,32 +305,57 @@ func progressivePerCluster(clusterGroupsMap clusterv1beta1.ClusterGroupsMap, len
 	}
 }
 
-func progressivePerGroup(clusterGroupsMap clusterv1beta1.ClusterGroupsMap, timeout time.Duration, statusFunc ClusterRolloutStatusFunc) RolloutResult {
-	rolloutClusters := map[string]ClusterRolloutStatus{}
-	timeoutClusters := map[string]ClusterRolloutStatus{}
+func progressivePerGroup(clusterGroupsMap clusterv1beta1.ClusterGroupsMap, timeout time.Duration, existingClusterStatus []ClusterRolloutStatus) RolloutResult {
+	rolloutClusters := []ClusterRolloutStatus{}
+	timeoutClusters := []ClusterRolloutStatus{}
+	existingClusters := make(map[string]bool)
 
-	clusterGroupKeys := clusterGroupsMap.GetOrderedGroupKeys()
+	for _, status := range existingClusterStatus {
+		if status.ClusterName == "" {
+			continue
+		}
+
+		existingClusters[status.ClusterName] = true
+		switch status.Status {
+		case ToApply:
+			// Set as false to consider the cluster in the decisionGroups iteration.
+			existingClusters[status.ClusterName] = false
+		case Failed, Progressing:
+			newStatus, needToRollout := determineRolloutStatusAndContinue(status, timeout)
```

Comment:
Can here combine the switch case and determineRolloutStatusAndContinue's switch case into one function?

Reply:
okay, done.